homework-submission/todo_react/src/ItemCreate.js

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import Delete from "./Delete";
+import Edit from "./Edit";
+import DatePicker from 'react-date-picker';
+// import DateP from "./DatePicker";
+
+export default class ItemCreate extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            to_do: [],
+            description: [],
+            deadline: [new Date()]
+        };
+    }
+
+    handleCreate = (e) => {
+        let newState = this.state
+        newState.to_do.push(this.state.description)
+        this.setState(newState)
+        //clean input
+        this.input.value = ''
+        e.preventDefault()
+    }
+
+    inputChange = (e) => { this.setState({ description: e.target.value }) }
+
+    handleDelete = (item) => {
+        let newState = this.state.to_do
+        newState.splice(item, 1)
+        this.setState({ newState })
+    }
+
+    handleUpdate = (e) => { this.setState({ to_do: [e.target.value] }) }
+
+    handleEdit = (e) => {
+        this.setState({
+            to_do: [<div id='edit'><input type="text" value={this.state.to_do} onChange={this.handleUpdate} />
```

Comment:
Never put an elements inside of a state!

Why? an element is a special kind of an object, and it has a big enough amount of properties, so if you want to change the property from any where in your project you will have to change it using the `state` or never.

This is a very small reason, you may learn more about that in the future.